src/main/java/uk/gov/pay/directdebit/mandate/services/MandateUpdateService.java

```diff
@@ -0,0 +1,34 @@
+package uk.gov.pay.directdebit.mandate.services;
+
+import uk.gov.pay.directdebit.gatewayaccounts.model.PaymentProvider;
+import uk.gov.pay.directdebit.mandate.dao.MandateDao;
+import uk.gov.pay.directdebit.mandate.model.MandateLookupKey;
+import uk.gov.pay.directdebit.mandate.model.MandateState;
+import uk.gov.pay.directdebit.mandate.model.PaymentProviderMandateId;
+import uk.gov.pay.directdebit.mandate.model.SandboxMandateId;
+import uk.gov.pay.directdebit.payments.model.GoCardlessMandateIdAndOrganisationId;
+
+import javax.inject.Inject;
+
+public class MandateUpdateService {
+
+    private final MandateDao mandateDao;
+
+    @Inject
+    public MandateUpdateService(MandateDao mandateDao) {
+        this.mandateDao = mandateDao;
+    }
+
+    public int updateStateByPaymentProviderMandateId(PaymentProvider paymentProvider, MandateLookupKey mandateLookupKey, MandateState mandateState) {
+        if (mandateLookupKey.getClass() == GoCardlessMandateIdAndOrganisationId.class) {
```

Comment:
The switching logic, albeit non-polymorphic as you point out is nicely contained in this single method.